pull.go

```diff
@@ -152,60 +152,60 @@ func localImageNameForReference(ctx context.Context, store storage.Store, srcRef
 
 // Pull copies the contents of the image from somewhere else to local storage.
 func Pull(ctx context.Context, imageName string, options PullOptions) error {
-	spec := imageName
 	systemContext := getSystemContext(options.SystemContext, options.SignaturePolicyPath)
-	srcRef, err := alltransports.ParseImageName(spec)
-	if err != nil {
-		if options.Transport == "" {
-			options.Transport = util.DefaultTransport
-		}
-		logrus.Debugf("error parsing image name %q, trying with transport %q: %v", spec, options.Transport, err)
-		transport := options.Transport
-		if transport != util.DefaultTransport {
-			transport = transport + ":"
-		}
-		spec = transport + spec
-		srcRef2, err2 := alltransports.ParseImageName(spec)
-		if err2 != nil {
-			return errors.Wrapf(err2, "error parsing image name %q", imageName)
-		}
-		srcRef = srcRef2
+
+	transport := options.Transport
+	if transport == "" {
+		transport = util.DefaultTransport
+	}
+	boptions := BuilderOptions{
+		FromImage:           imageName,
+		Transport:           transport,
+		SignaturePolicyPath: options.SignaturePolicyPath,
+		SystemContext:       systemContext,
+		PullBlobDirectory:   options.BlobDirectory,
 	}
+
 	if options.Quiet {
-		options.ReportWriter = nil // Turns off logging output
+		boptions.ReportWriter = nil // Turns off logging output
 	}
-	var names []string
+	srcRef, _, err := resolveImage(ctx, systemContext, options.Store, boptions)
+	if err != nil {
+		return err
+	}
+
+	var errs *multierror.Error
 	if options.AllTags {
 		if srcRef.DockerReference() == nil {
 			return errors.New("Non-docker transport is currently not supported")
 		}
+
+		spec := transport + srcRef.DockerReference().Name()
```

Comment:
Looking at `resolveName`, the transport value does not include the `:` separator unless it is `docker://` (or `DefaultTransport`, depending on how exactly it is hard-coded); it needs to be added.

Or, well, if you want to hard-code `docker.Transport`, just do so.

Reply:
I would like to get rid of this code altogether and get it into a function provided by containers/image written by you or @vrothberg.  But for now, I would just like to fix the issue of buildah pull not working like buildah from.

Reply:
\*shrug\* I don’t quite buy “this is an important feature and that justifies not getting the details right”, but in the end, I’m just pointing out breakage I am noticing; I am not a buildah maintainer and I won’t be approving the PR.

Reply:
(To be clear, I am not saying that consolidation of the `transport + name` logic into a function is a requirement of this PR; the goal here was getting the `":"` handling right, and consolidation was a side suggestion.)